var/spack/repos/builtin/packages/apcomp/package.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+import os
+import socket
+import llnl.util.tty as tty
+
+
+def cmake_cache_entry(name, value, vtype=None):
+    """
+    Helper that creates CMake cache entry strings used in
+    'host-config' files.
+    """
+    if vtype is None:
+        if value == "ON" or value == "OFF":
+            vtype = "BOOL"
+        else:
+            vtype = "PATH"
+    return 'set({0} "{1}" CACHE {2} "")\n\n'.format(name, value, vtype)
+
+
+class Apcomp(Package):
+    """A multi use-case image compositor"""
+
+    homepage = 'https://github.com/Alpine-DAV/ap_compositor'
+    git      = 'https://github.com/Alpine-DAV/ap_compositor.git'
+    url      = "https://github.com/Alpine-DAV/ap_compositor/releases/download/v0.0.1/apcomp-v0.0.1.tar.gz"
+
+    maintainers = ['mclarsen', 'cyrush']
+
+    version('master', branch='master', submodules='True')
+    version('0.0.1', sha256="cbf85fe58d5d5bc2f468d081386cc8b79861046b3bb7e966edfa3f8e95b998b2", preferred=True)
+
+    variant('openmp', default=True, description='Build with openmp support')
+    variant('mpi', default=True, description='Build with openmp support')
+    variant('shared', default=True, description='Build Shared Library')
+
+    depends_on('cmake@3.9:', type='build')
+    depends_on("mpi", when="+mpi")
+
+    root_cmakelists_dir = 'src'
+
+    def install(self, spec, prefix):
+        """
+        Build and install APComp
+        """
+        with working_dir('spack-build', create=True):
+            py_site_pkgs_dir = None
```

Comment:
Is there a reason you initialize this to `None` to pass it to `create_host_config` vs. just passing `None` to `create_host_config` directly?

Also, while `create_host_config` takes a `py_site_pkgs_dir` parameter, it doesn't do anything with that argument. Is this preparing for some future PR?

Reply:
this is a hold over from using another package as a starting point, we can prune.